core/src/test/kotlin/org/stellar/anchor/sep31/Sep31ServiceTest.kt

```diff
@@ -715,35 +814,138 @@ class Sep31ServiceTest {
-          "type" to "1",
-          "receiver_routing_number" to "SWIFT",
-        ),
-      )
+        Sep31TxnFields(
+            hashMapOf(
+                "receiver_account_number" to "1",
+                "type" to "1",
+                "receiver_routing_number" to "SWIFT",
+            ),
+        )
 
     // Provide fee response.
     every { feeIntegration.getFee(any()) } returns
-      GetFeeResponse(
-        Amount(
-          "2",
-          "stellar:USDC",
-        ),
-      )
+        GetFeeResponse(
+            Amount(
+                "2",
+                "stellar:USDC",
+            ),
+        )
 
     // POST transaction
     val jwtToken = TestHelper.createJwtToken()
     var gotResponse: Sep31PostTransactionResponse? = null
     assertDoesNotThrow { gotResponse = sep31Service.postTransaction(jwtToken, postTxRequest) }
 
     val wantResponse =
-      Sep31PostTransactionResponse.builder()
-        .id(gotResponse!!.id)
-        .stellarAccountId("GA7FYRB5VREZKOBIIKHG5AVTPFGWUBPOBF7LTYG4GTMFVIOOD2DWAL7I")
-        .stellarMemo("")
-        .stellarMemoType("")
-        .build()
+        Sep31PostTransactionResponse.builder()
+            .id(gotResponse!!.id)
+            .stellarAccountId("GA7FYRB5VREZKOBIIKHG5AVTPFGWUBPOBF7LTYG4GTMFVIOOD2DWAL7I")
+            .stellarMemo("")
+            .stellarMemoType("")
+            .build()
     assertEquals(wantResponse, gotResponse)
   }
+
+  @Test
+  fun test_info_response() {
+    val info = sep31Service.info
+    val jpyc = info.receive.get("JPYC")!!
+    val usdc = info.receive.get("USDC")!!
+
+    // Test correctness of reading test_assets.json
+    assertTrue(jpyc.enabled)
+    assertTrue(jpyc.quotesSupported)
+    assertTrue(jpyc.quotesRequired)
+    assertEquals(0, jpyc.feeFixed)
+    assertEquals(0, jpyc.feePercent)
+    assertEquals(1, jpyc.minAmount)
+    assertEquals(1000000, jpyc.maxAmount)
+    assertNotNull(jpyc.sep12)
+    assertNotNull(jpyc.sep12.sender)
+    assertNotNull(jpyc.sep12.sender.types)
+    assertEquals(1, jpyc.sep12.sender.types.size)
+    assertNotNull(jpyc.sep12.receiver)
+    assertNotNull(jpyc.sep12.receiver.types)
+    assertEquals(1, jpyc.sep12.receiver.types.size)
+    assertNotNull(jpyc.fields.transaction)
+    assertEquals(3, jpyc.fields.transaction.size)
+
+    assertTrue(usdc.enabled)
+    assertTrue(usdc.quotesSupported)
+    assertTrue(usdc.quotesRequired)
+    assertEquals(0, usdc.feeFixed)
+    assertEquals(0, usdc.feePercent)
+    assertEquals(1, usdc.minAmount)
+    assertEquals(1000000, usdc.maxAmount)
+    assertNotNull(usdc.sep12)
+    assertNotNull(usdc.sep12.sender)
+    assertNotNull(usdc.sep12.sender.types)
+    assertEquals(3, usdc.sep12.sender.types.size)
+    assertNotNull(usdc.sep12.receiver)
+    assertNotNull(usdc.sep12.receiver.types)
+    assertEquals(2, usdc.sep12.receiver.types.size)
+    assertNotNull(usdc.fields.transaction)
+    assertEquals(3, usdc.fields.transaction.size)
```

Comment:
💡  As I said in a previous comment, I prefer comparing whole objects because it's easier to identify changes in the object schema that way.

When we compare individual fields like you're doing here, we don't catch inequalities in additional fields. Additional fields should break our tests for us to keep the tests up to date ✅

Reply:
Updated.